FloripaPublicTransportation/RoutesListViewController.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  RoutesListViewController.swift
+//  FloripaPublicTransportation
+//
+//  Created by Marcelo Gobetti on 2/19/16.
+//  Copyright © 2016 Marcelo Gobetti. All rights reserved.
+//
+
+import UIKit
+
+class RoutesListViewController: UITableViewController, UISearchBarDelegate {
+    // MARK: Public properties
+    
+    private var _streetToSearch: String? // stored property
+    var streetToSearch: String? { // public computed property
+        get { return _streetToSearch }
+        set {
+            guard newValue != nil else {
+                return
+            }
+            
+            _streetToSearch = newValue
+            
+            self.searchBar?.text = _streetToSearch
+            
+            if self.routes?.count > 0 {
+                self.routes?.removeAll()
+            }
+            
+            guard Connectivity.isConnectedToNetwork() else {
+                Connectivity.popNoConnectionAlert(self)
+                return
+            }
+            
+            self.activityIndicator.startAnimating()
+            self.view.bringSubviewToFront(self.activityIndicator)
+            
+            RestApi.findRoutesByStopName(_streetToSearch!) { routes in
+                self.routes = routes
+                dispatch_async(dispatch_get_main_queue(), {
+                    self.activityIndicator.stopAnimating()
+                })
+            }
+        }
+    }
+    
+    // MARK: Private properties
+    
+    @IBOutlet private weak var searchBar: UISearchBar?
+    private var activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: UIActivityIndicatorViewStyle.Gray)
+    
+    private var _routes: [Route]? // stored property
+    private var routes: [Route]? { // computed property
+        get { return _routes }
+        set {
+            _routes = newValue
```

Comment:
Maybe using delegates and callbacks, however I prefer the property setter. IMO this is the reason why properties exist, after all. Without properties I could have manually written a SetRoutes function that would set the routes field and do something with it.